service/rpc/handler.go

```diff
@@ -0,0 +1,25 @@
+package rpc
+
+import (
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/celestiaorg/celestia-node/service/header"
+	"github.com/celestiaorg/celestia-node/service/share"
+	"github.com/celestiaorg/celestia-node/service/state"
+)
+
+var log = logging.Logger("rpc")
+
+type Handler struct {
+	state  *state.Service
+	share  share.Service
+	header *header.Service
+}
+
+func NewHandler(state *state.Service, share share.Service, header *header.Service) *Handler {
+	return &Handler{
+		state:  state,
+		share:  share,
+		header: header,
+	}
+}
```

Comment:
design nit: This kind of struct does not do anything meaningful and does not have any state, so it does really make sense to define it as a struct, and a functional approach to registering handlers would like much simpler, elegant, and readable. Example:
```go
func RegisterBalanceHandler(rpc *Server, state *state.Service) {
	rpc.RegisterHandlerFunc(balanceEndpoint, func(w http.ResponseWriter, r *http.Request) {
	...
	state.Balance()
	..
	} , http.MethodGet)
}
```

This way, you could also simplify testing by testing each Handler atomically.

Reply:
Surely, no need to refactor everything right now, JFYI

Reply:
Leaving as is for now bc assuming we will refactor this later. Thx tho